src/lib.rs

```diff
@@ -315,6 +316,17 @@ impl VadSession {
                                 - self.silent_samples)
                                 / (self.config.sample_rate / 1000);
 
+                            // Since speech_end_ms does not include silent_samples, the post
+                            // padding should be added to speech_end_ms.
+                            let speech_end_with_pad_ms =
+                                speech_end_ms + self.config.post_speech_pad.as_millis() as usize;
+                            // TODO: Just some quick double checks, remember to delete these before git merge.
```

Comment:
:eyes:

Reply:
I'll keep them here for a while so have more confidence the modified logic is correct. Will delete these checks before merging.